tasks/services/works.js

```diff
@@ -36,26 +79,101 @@ function createCastOrderIntentTaskPlan (hook) {
   const completeOrderTaskPlanId = hook.data.taskPlanId
 
   return taskPlans.get(completeOrderTaskPlanId)
-  .then((completeOrderTaskPlan) => {
-    const groupId = completeOrderTaskPlan.params.consumerAgentId
-    const orderId = completeOrderTaskPlan.params.orderId
-    const supplierAgentId = completeOrderTaskPlan.params.supplierAgentId
-    const params = { orderId, consumerAgentId: groupId, supplierAgentId }
-
-    return relationships.find({
-      query: {
-        sourceId: groupId,
-        relationshipType: 'member'
+    .then((completeOrderTaskPlan) => {
+      const groupId = completeOrderTaskPlan.params.consumerAgentId
+      const orderId = completeOrderTaskPlan.params.orderId
+      const supplierAgentId = completeOrderTaskPlan.params.supplierAgentId
+      const params = { orderId, consumerAgentId: groupId, supplierAgentId }
+
+      return relationships.find({
+        query: {
+          sourceId: groupId,
+          relationshipType: 'member'
+        }
+      })
+        .then((relationships) => {
```

Comment:
ramda golf

```js
.then(pipe(
  map(pipe(
    prop('targetId'),
    assigneeId => taskPlans.create({
      assigneeId,
      taskRecipeId,
      params
    })
  )),
  Promise.all
))
```

Reply:
oooh shiny :) needs to change a bit though now `[params, relationships]` are the args to that `.then` function...